fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryScanPlan.java

```diff
@@ -173,16 +203,50 @@ public int getComplexity() {
      * @return the rewritten planner graph that models scanned storage as a separate node that is connected to the
      *         actual scan plan node.
      */
+    @SuppressWarnings("UnstableApiUsage")
     @Nonnull
     @Override
     public PlannerGraph rewritePlannerGraph(@Nonnull List<? extends PlannerGraph> childGraphs) {
         Verify.verify(childGraphs.isEmpty());
-        final PlannerGraph.Node root = new PlannerGraph.Node(this,
-                getClass().getSimpleName());
-        final PlannerGraph.SourceNode source = new PlannerGraph.SourceNode(getRange());
-        return PlannerGraph.builder(root)
-                .addNode(source)
-                .addEdge(source, root, new PlannerGraph.Edge())
-                .build();
+
+        final Optional<TupleRange> tupleRangeOptional = getTupleRangeOptional();
+
+        final ImmutableList.Builder<String> detailsBuilder = ImmutableList.builder();
+        final ImmutableMap.Builder<String, Attribute> additionalAttributes = ImmutableMap.builder();
+
+        if (tupleRangeOptional.isPresent()) {
+            final TupleRange tupleRange = tupleRangeOptional.get();
```

Comment:
This tuple range handling code seems to be the same as the code for the index range scans (which makes sense, since they use the same implementation). Could you pull this out into a utility method (possibly on `ScanComparisons`)?